src/scippneutron/file_loading/nxdetector.py

```diff
@@ -6,13 +11,30 @@
 from ._common import to_plain_index
 
 
+class EventSelector:
+    """A proxy object for creating and NXdetector based on a selection of events.
+    """
+    def __init__(self, detector):
+        self._detector = detector
+
+    def __getitem__(self, select) -> NXdetector:
```

Comment:
```suggestion
    def __getitem__(self, select: ScippIndex) -> NXdetector:
```
? Or is the type different here?